### ISSUE TITLE
Error handling for database errors in get leads function

### ISSUE THREAD

**Issue by galbwe** [codefordenver/partner-finder#75]
Sending a request to `POST /leads` with a body that is missing a `"company_name"` field results in a 500 response because there is a not null constraint on the `company_name` column of the leads table in postgres. The api should instead return a 400 response and an error message because this was a user error.

Add error handling to the database calls in `_get_all_leads` in `leads.py`. Catch the appropriate database error. If a not null contstraint was violated, return a 400 response.

**Comment by Nova791** [comment]
![](https://github.trello.services/images/mini-trello-icon.png) [Error handling for Database Errors](https://trello.com/c/SfGcmtEp/5-error-handling-for-database-errors)

**Comment by yimjay** [comment]
`POST /leads` returns error 400 with message when user tries to create a new lead with no `"company_name"` field. Error handling logic was added to `_create_new_lead` rather than `_get_all_leads` however

![image](https://user-images.githubusercontent.com/13406224/130531481-c1252309-e269-47f2-9a81-b7fab6149ed2.png)

**Comment by galbwe** [comment]
@yimjay here is a request body that produces the 500 error. My apologies for not including it to begin with. It should return a 400 or a 422

```
{
  "assigned": "Jeeves",
  "colorado_region": "Denver",
  "company_address": "5090 S Quebec St, Denver, CO 80237",
  "contact_name": "Mr. Ronald",
  "county": "Denver",
  "data_source": "socrata",
  "email": "ronald@mcdonalds.com"
}
```

and here is the error message:

```
api_1       | sqlalchemy.exc.IntegrityError: (psycopg2.errors.NotNullViolation) null value in column "company_name" of relation "leads" violates not-null constraint
api_1       | DETAIL:  Failing row contains (3190, null, 5090 S Quebec St, Denver, CO 80237, Mr. Ronald, null, null, null, ronald@mcdonalds.com, null, null, null, null, null, null, null, null, null, null, null, null, null, Denver, Denver, socrata, Jeeves).
api_1       | 
api_1       | [SQL: 
api_1       |                 INSERT INTO leads (assigned,colorado_region,company_address,contact_name,county,data_source,email)
api_1       |                 VALUES (%(assigned)s,%(colorado_region)s,%(company_address)s,%(contact_name)s,%(county)s,%(data_source)s,%(email)s)
api_1       |                 RETURNING *;
api_1       |                 ]
api_1       | [parameters: {'assigned': 'Jeeves', 'colorado_region': 'Denver', 'company_address': '5090 S Quebec St, Denver, CO 80237', 'contact_name': 'Mr. Ronald', 'county': 'Denver', 'data_source': 'socrata', 'email': 'ronald@mcdonalds.com'}]
api_1       | (Background on this error at: http://sqlalche.me/e/14/gkpj)
```

**Comment by galbwe** [comment]
addressed by #94
